server/handler/check_run_create_hander.go

```diff
@@ -0,0 +1,110 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/mattermost/release-bot/config"
+	"github.com/mattermost/release-bot/log"
+	"github.com/mattermost/release-bot/store"
+	"github.com/palantir/go-githubapp/githubapp"
+)
+
+type checkRunCreateHandler struct {
+	LogType           string
+	Config            *config.Config
+	ClientCreator     githubapp.ClientCreator
+	EventContextStore store.EventContextStore
+}
+
+type CheckRunCreateRequest struct {
+	Token   string `json:"token"`
+	Name    string `json:"name"`
+	Title   string `json:"title"`
+	Summary string `json:"summary"`
+}
+
+func (ccr *CheckRunCreateRequest) Log() {
+	log.Info("create_check_run").
+		Str("name", ccr.Name).
+		Str("title", ccr.Title).
+		Str("summary", ccr.Summary).
+		Msg("")
+}
+
+type CreateCheckRunResponse struct {
+	ID *int64 `json:"id"`
+}
+
+func (vh *checkRunCreateHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	log.Info(vh.LogType).Msg("Create Check Run Request is received!")
+	if r.Method != "POST" {
+		http.Error(w, "Only POST method is supported!", http.StatusMethodNotAllowed)
+		return
+	}
+	var createCheckRunRequest CheckRunCreateRequest
+
+	err := json.NewDecoder(r.Body).Decode(&createCheckRunRequest)
+	if err != nil {
+		log.Info(vh.LogType).
```

Comment:
more like a preference. If you handle an error I think there's no need to log it. Also the level should be `error` instead of info. Because it's clear that it's a bad request I wouldn't log it. I think it's a good idea to return back to the API caller a payload with the errors

Reply:
The logging is needed, since it will allow us to create an alert . So we will be notified for the pipelines which sends invalid request to create check. Will change it to error.

Reply:
Changed log level to error.